### PR TITLE
Refacto 280

### PR DESCRIPTION
**Pull request by sbarguil**
#280

### CODE REVIEW

File: yang/ietf-l3vpn-ntw.yang [hunk 1]
```diff
@@ -753,38 +753,34 @@ module ietf-l3vpn-ntw {
...
                   description
                     "Encapsulation types.";
-                  leaf encapsulation-type {
-                    type identityref {
-                      base vpn-common:encapsulation-type;
+                  leaf connection-type {
+                    type enumeration {
+                        enum LOCAL {
+                          description
+                            "evpn bgp-instance is operationally up";
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:771]: Why are we talking about EVPN/operationally?

I'm not sure to get the intent here.

File: yang/ietf-l3vpn-ntw.yang [hunk 2]
```diff
@@ -753,38 +753,34 @@ module ietf-l3vpn-ntw {
...
+                        }
+                        enum REMOTE {
+                          description
+                            "evpn bgp-instance is operationally down";
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:775]: Idem as above.

File: yang/ietf-l3vpn-ntw.yang [hunk 3]
```diff
@@ -979,6 +892,34 @@ module ietf-l3vpn-ntw {
                           "SDP/Far End/LDP neighbour reference.";
                       }
                     }
+                    container vxlan {
+                        description
+                          "Only applies when the type of the
+                           tagged interface is 'vxlan'.";
+                      if-feature "vpn-common:vxlan";
+                      description
+                        "QinQ.";
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:901]:
```suggestion
                        "VXLAN.";
```

File: yang/ietf-l3vpn-ntw.yang [hunk 4]
```diff
@@ -753,38 +753,34 @@ module ietf-l3vpn-ntw {
...
                     default "vpn-common:untagged-int";
                     description
                       "Encapsulation type. By default, the encapsulation
                        type is set to 'untagged'.";
                   }
-                  container logical-interface {
-                    description
-                      "Reference of a logical interface
-                       type.";
-                    leaf local-reference {
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:773]: We agreed to have this one to address #266

File: yang/ietf-l3vpn-ntw.yang [hunk 5]
```diff
@@ -753,38 +753,34 @@ module ietf-l3vpn-ntw {
                   description
                     "An identifier of an active VPN instance profile.";
                 }
+                leaf bearer-reference {
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:756]: It makes sense to make maintained under connection to group < 3 connectivity parameters.

File: yang/ietf-l3vpn-ntw.yang [hunk 6]
```diff
@@ -872,75 +849,11 @@ module ietf-l3vpn-ntw {
...
-                      }
-                      leaf-list peer-ip-address {
-                        type inet:ip-address;
-                        description
-                          "List of peer's IP addresses.";
-                      }
-                    }
                   }
-                  container bearer {
+                  container remote-endpoints {
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:853]: The proposed name is confusing. 

This is more a l2 termination. We may call it transport-logical-interface or something similar.

**Review by boucadair — CHANGES_REQUESTED** [review]

File: yang/ietf-l3vpn-ntw.yang (hunk 1 above)
Comment (sbarguil) [reply yang/ietf-l3vpn-ntw.yang:771]: Sorry I have copy/paste from other model. My mistake

File: yang/ietf-l3vpn-ntw.yang (hunk 2 above)
Comment (sbarguil) [reply yang/ietf-l3vpn-ntw.yang:775]: Sorry I have copy/paste from other model. My mistake

File: yang/ietf-l3vpn-ntw.yang (hunk 4 above)
Comment (sbarguil) [reply yang/ietf-l3vpn-ntw.yang]: reverted 45ca37511f6f7c0b0d19c1f3affe9d742a788e28

File: yang/ietf-l3vpn-ntw.yang [hunk 7]
```diff
@@ -753,15 +753,28 @@ module ietf-l3vpn-ntw {
...
                   description
                     "Encapsulation types.";
-                  leaf encapsulation-type {
-                    type identityref {
-                      base vpn-common:encapsulation-type;
+                  leaf connection-type {
+                    type enumeration {
+                        enum LOCAL {
+                          description
+                            "Local termination of the VPN network access.";
```
Comment (boucadair) [inline yang/ietf-l3vpn-ntw.yang:771]: Thanks Samier for updating the description. 

I still don't get this. What is a local/remote termination?

**Comment by boucadair** [comment]
Merged as discussed in our conf call. Will update and tweak the text a little bit.

**Pull request merged**